va/va_fei_h264.h

```diff
@@ -0,0 +1,396 @@
+     * each MB control data structure is defined by VAEncFEIMBControlH264.
+     * Buffer size shall not be less than the number of 16x16 blocks multiplied by
+     * sizeof(VAEncFEIMBControlH264).
+     * Note: if mb_qp is set, VAEncQpBufferH264 is expected.
+     */
+    VABufferID    mb_ctrl;
+    /** \brief distortion output of MB ENC or ENC_PAK.
+     * Each 16x16 block has one distortion data with VAEncFEIDistortionH264 layout
+     * Buffer size shall not be less than the number of 16x16 blocks multiplied by
+     * sizeof(VAEncFEIDistortionH264).
+     */
+    VABufferID    distortion;
+    /** \brief MVs data output of MB ENC.
+     * Each 16x16 block has one MVs data with layout VAMotionVector
+     * Buffer size shall not be less than the number of 16x16 blocks multiplied by
+     * sizeof(VAMotionVector) * 16.
+     */
+    VABufferID    mv_data;
+    /** \brief MBCode data output of MB ENC.
+     * Each 16x16 block has one MB Code data with layout VAEncFEIMBCodeH264
+     * Buffer size shall not be less than the number of 16x16 blocks multiplied by
+     * sizeof(VAEncFEIMBCodeH264).
+     */
+    VABufferID    mb_code_data;
+    /** \brief Qp input buffer with layout VAEncQpBufferH264. It is valid only when mb_qp is set to 1.
+     * The data in this buffer correspond to the input source.
+     * One Qp per 16x16 block in raster scan order, each Qp is a signed char (8-bit) value.
+     **/
+    VABufferID    qp;
+    /** \brief MV predictor. It is valid only when mv_predictor_enable is set to 1.
+     * Each 16x16 block has one or more pair of motion vectors and the corresponding
+     * reference indexes as defined by VAEncFEIMVPredictorH264. 16x16 block is in raster scan order.
+     * Buffer size shall not be less than the number of 16x16 blocks multiplied by
+     * sizeof(VAEncFEIMVPredictorH264). */
+    VABufferID    mv_predictor;
+
+    /** \brief number of MV predictors. It must not be greater than maximum supported MV predictor. */
+    uint32_t      num_mv_predictors_l0      : 16;
+    uint32_t      num_mv_predictors_l1      : 16;
+
+    /** \brief control parameters */
+    uint32_t      search_path               : 8;
+    uint32_t      len_sp                    : 8;
+    uint32_t      reserved0	                : 16;
+
+    uint32_t      sub_mb_part_mask          : 7;
+    uint32_t      intra_part_mask           : 5;
+    uint32_t      multi_pred_l0             : 1;
+    uint32_t      multi_pred_l1             : 1;
+    uint32_t      sub_pel_mode              : 2;
+    uint32_t      inter_sad 	            : 2;
+    uint32_t      intra_sad                 : 2;
+    uint32_t      distortion_type           : 1;
+    uint32_t      repartition_check_enable  : 1;
+    uint32_t      adaptive_search           : 1;
+    uint32_t      mv_predictor_enable       : 1;
+    uint32_t      mb_qp                     : 1;
+    uint32_t      mb_input                  : 1;
+    uint32_t      mb_size_ctrl              : 1;
```

Comment:
Hi Sreerenjb, 
These input enabling bits are  only for more obvious info to customers, they definitely could be gotten by checking buffer_ids are valid or invalid, but we hope it's clear input enabling or disabling without checking.
For output buffers mb_code_data/mv_data/distortion, they are really no enabling bits, and driver check if they are enabled by this way.